lib/webdrivers/common.rb

```diff
@@ -7,40 +7,52 @@ class << self
       attr_accessor :version
 
       def update
+        if correct_binary? # Already have desired or latest version
+          Webdrivers.logger.debug 'Expected webdriver version found'
+          return binary
+        end
+
+        # If site is down
         unless site_available?
-          return current_version.nil? ? nil : binary
+          # No existing binary and we can't download
+          raise StandardError, update_failed_msg if current_version.nil?
+
+          # Use existing binary
+          Webdrivers.logger.error "Can not reach update site. Using existing #{file_name} #{current_version}"
+          return binary
         end
 
         # Newer not specified or latest not found, so use existing
         return binary if desired_version.nil? && File.exist?(binary)
 
         # Can't find desired and no existing binary
-        if desired_version.nil?
-          msg = "Unable to find the latest version of #{file_name}; try downloading manually from #{base_url} and place in #{install_dir}"
-          raise StandardError, msg
-        end
-
-        if correct_binary?
-          Webdrivers.logger.debug 'Expected webdriver version found'
-          return binary
-        end
+        raise StandardError, update_failed_msg if desired_version.nil?
 
         remove # Remove outdated exe
-        download # Fetch latest
+        download # Fetch desired or latest
       end
 
       def desired_version
-        if version.is_a?(Gem::Version)
-          version
-        elsif version.nil?
-          latest_version
-        else
-          Gem::Version.new(version.to_s)
-        end
+        ver = if version.is_a?(Gem::Version)
+                version
+              elsif version.nil?
+                latest_version
+              else
+                Gem::Version.new(version.to_s)
+              end
+
+        Webdrivers.logger.debug "Desired version: #{ver}"
+        ver
       end
 
       def latest_version
-        raise StandardError, 'Can not reach site' unless site_available?
+        unless site_available?
```

Comment:
Let's swap this around; put this at the beginning:
```
return downloads.keys.max if site_available?
```
That'll let you remove the the unless block.